vault/diagnose/tls_verification_test.go

```diff
@@ -146,9 +147,13 @@ func TestTLSMismatchedCryptographicInfo(t *testing.T) {
 	if errs == nil || len(errs) != 1 {
 		t.Fatalf("TLS Config check on fake certificate should fail")
 	}
-	if !strings.Contains(errs[0].Error(), "tls: private key type does not match public key type") {
+	if !strings.Contains(errs[0].Error(), "certificate signed by unknown authority") {
 		t.Fatalf("Bad error message: %s", errs[0])
 	}
+	_, err := tls.LoadX509KeyPair("./../../api/test-fixtures/keys/cert.pem", "./test-fixtures/ecdsa.key")
```

Comment:
The ListenerChecks returns with error before it tls.LoadX509KeyPair in TLSFileChecks